tenets/codelingo/code-review-comments/structs-with-mutex/codelingo.yaml

```
@@ -0,0 +1,33 @@
+tenets:
+  - name: structs-with-mutex
+    flows:
+      codelingo/review:
+        comment: If the receiver is a struct that contains a sync.Mutex or similar synchronizing field, the receiver must be a pointer to avoid copying. From [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type)
+      codelingo/docs:
+        title: Structs with Mutex
+        body: If the receiver is a struct that contains a sync.Mutex or similar synchronizing field, the receiver must be a pointer to avoid copying.
+    query: |
+      import codelingo/ast/go
+
+      
+      go.type_spec(depth = any):
+        go.ident:
+          type as structType
+        go.struct_type:
+          go.field_list:
+            go.field:
+              go.names:
+                go.ident
+              go.selector_expr:
+                go.ident:
+                  name == "sync"
+                go.ident:
+                  name == "Mutex"
+      go.func_decl(depth = any):
+        go.field_list:
+          go.field:
+            go.names:
+              go.ident
+            @review comment
+            go.ident:
```

```suggestion
            go.ident: // ident is not a child of a star_expr, so it is not a pointer receiver
```
Do we need some kind of exclude here to confirm the receiver is not a pointer? I'm assuming that pointers receivers will have some kind binary expression in between the `go.field` and `go.ident` but this is not obvious. Comment should help.

We could make this even more explicit with `include/exclude` or `path`.

---

@BlakeMScurr Yeah, the ident would have a go.star_expr as a parent. That comment should be enough.